### PR TITLE
Create new `ReleaseTools` module

### LINKED ISSUE #3195: Release Automation Improvements
We should clean up/optimize the release automation for the VSCode extension in order to reduce the time dedicated to each release so that we can optimize our time for improving the extension and release at a more frequent cadence. 

cc: @rjmholt @andschwa

### PR DESCRIPTION
**Pull request by andschwa**
This PR started with some minor cleanups with me trying to delete old, unused code and then turned into me writing some a new `ReleaseTools` module for my needs as maintainer. Fixes #3195.

### CODE REVIEW

File: tools/ReleaseTools.psm1 [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        { $_.StartsWith("##") }, "SkipUntil"
+    ).Where(
+        { $_.StartsWith("## $($Release.tag_name)") }, "Until"
+    )
+}
+
+<#
+.SYNOPSIS
+  Updates the CHANGELOG file with PRs merged since the last release.
+.DESCRIPTION
```
Comment (andschwa) [inline tools/ReleaseTools.psm1:106]: TODO: Update this and comment to the other one.

File: tools/ReleaseTools.psm1 [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        if ($branch -ne "release/$version") {
+            git checkout -b "release/$version"
+        }
+        git add $ChangelogFile
+        git commit -m "Update CHANGELOG for $Version"
+    }
+    Pop-Location
+}
+
+function New-DraftRelease {
```
Comment (andschwa) [inline tools/ReleaseTools.psm1:172]: TODO: Test this part.

File: tools/ReleaseTools.psm1 [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    [CmdletBinding(SupportsShouldProcess)]
+    param(
+        [Parameter(Mandatory)]
+        [ValidateSet("vscode-powershell", "PowerShellEditorServices")]
+        [string]$RepositoryName,
+
+        [Parameter(Mandatory)]
+        [ValidateScript({ $_.StartsWith("v") })]
+        [string]$Version
```
Comment (andschwa) [inline tools/ReleaseTools.psm1:181]: TODO: Get version from changelog file so I don't have to repeat it here.

Comment (andschwa) [reply tools/ReleaseTools.psm1:182]: Tested, fixed! Nice.

File: tools/ReleaseTools.psm1 (hunk 2 above)
Comment (andschwa) [reply tools/ReleaseTools.psm1:172]: Tested, fixed! Nice.

File: tools/ReleaseTools.psm1 [hunk 4]
```diff
@@ -0,0 +1,194 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+#requires -Version 6.0
```
Comment (rjmholt) [inline tools/ReleaseTools.psm1:4]: 6.0 is defunct, may as well require 7.0
```suggestion
#requires -Version 7.0
```

**Review by rjmholt — APPROVED** [review]
Please rename the PR to something more descriptive, like "Clean up changelog and release scripts"

**Comment by andschwa** [comment]
> Please rename the PR to something more descriptive, like "Clean up changelog and release scripts"

@rjmholt TBF it _started_ as a minor cleanup, but then I did more 😅

**Pull request merged**